src/state.rs

```diff
@@ -1,11 +1,49 @@
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::Addr;
 use cw_storage_plus::Item;
+use neutron_sdk::sudo::msg::RequestPacket;
 
 pub const CONFIG: Item<Config> = Item::new("config");
 
+pub const INTERCHAIN_ACCOUNT: Item<Option<InterchainAccount>> = Item::new("interchain_account");
+
+// if true, interchain operation is in progress and we cannot make an operation
+pub const INTERCHAIN_TX_IN_PROGRESS: Item<bool> = Item::new("interchain_tx_in_progress");
+
+// to understand what happened with IBC calls
+pub const IBC_CALLBACK_STATES: Item<Vec<IbcCallbackState>> = Item::new("ibc_callback_states");
+
 #[cw_serde]
 pub struct Config {
-    // only main dao can execute Transfer msg
-    pub main_dao: Addr,
+    pub connection_id: String,
+    pub transfer_channel_id: String,
+    pub ibc_neutron_denom: String,
+    pub ibc_timeout_seconds: u64,
+}
+
+#[cw_serde]
+pub struct InterchainAccount {
+    // ica address on remote network
+    pub address: String,
+    pub port_id: String,
+    pub channel_id: String,
+    pub counterparty_channel_id: String,
+}
+
+#[cw_serde]
+pub struct OpenAckVersion {
+    pub version: String,
+    pub controller_connection_id: String,
+    pub host_connection_id: String,
+    pub address: String,
+    pub encoding: String,
+    pub tx_type: String,
+}
+
+#[cw_serde]
+pub enum IbcCallbackState {
+    Response(RequestPacket, u64),      // request_packet, block_height
+    Timeout(RequestPacket, u64),       // request_packet, block_height
+    Error(RequestPacket, String, u64), // error with request_packet, details, block_height
+
+    OpenAckError(String, u64, String, String, String, String), // parse_error, block_height, port_id, channel_id, counterparty_channel_id, counterparty_version
 }
```

Comment:
afaik the structure of `IBC_CALLBACK_STATES` will be as array of arrays of 2-6 elements. given the fact there is 2 elements for both `Response` and `Timeout`, we won't be able to distinguish between them when reading query response.

like that. what is what here?
```
[
    [{packet}, 64],
    [{packet}, 66]
]
```

please prove me wrong or replace the enum variants with structures

Reply:
If I understand you correctly, no. Yo'll be able to differentiate between timeout or response. Structure is like this:
```
[
    Response(packet, u64],
    Timeout(packet, u64],
]
```

(example of check in tests https://github.com/neutron-org/neutron-airdrop-transfer/pull/3/files#diff-4b10913f4c3e7858207826d4e2e777308bebc273b75b7e6443880d8181111fb8R146)

Reply:
okay, agree. seems it's not as I described by me, but as
```
[
    {
        response: [{packet}, height]
    },
    {
        timeout: [{packet}, height]
    }
]
```